### PR TITLE
[INLONG-9520][Sort] Upgrade sort format to support InLongMsg body containing multiple pieces of data

### LINKED ISSUE #9520: [Improve][Sort]Upgrade sort format to support InLongMsg body containing multiple pieces of data
### Description

_No response_

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
…le pieces of data (#9520)

### Prepare a Pull Request

- Title : Upgrade sort format to support InLongMsg body containing multiple pieces of data (#9520)

- Fixes #9520

### Motivation

Upgrade sort format to support InLongMsg body containing multiple pieces of data

### CODE REVIEW

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/util/LogCounter.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * A LogCounter is responsible for controlling the logging frequency.
+ */
+public class LogCounter implements Serializable {
+
+    private AtomicInteger counter = new AtomicInteger(0);
+
+    private int start = 10;
+    private int control = 1000;
```
Comment (EMsnap) [inline inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/util/LogCounter.java:31]: 60 * 1000 is redundant

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultTableFormatSerializer.java [hunk 2]
```diff
@@ -67,15 +52,14 @@ public void flatMap(
         try {
             bytes = serializationSchema.serialize(row);
         } catch (Exception e) {
-            if (ignoreErrors) {
```
Comment (EMsnap) [inline inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultTableFormatSerializer.java:70]: sorry why remove this ignore errors?

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatConstants.java [hunk 3]
```diff
@@ -38,6 +39,11 @@ public class TableFormatConstants {
     public static final char DEFAULT_DELIMITER = ',';
     public static final char DEFAULT_ENTRY_DELIMITER = '&';
     public static final char DEFAULT_KV_DELIMITER = '=';
+    public static final Character DEFAULT_LINE_DELIMITER = null;
+    public static final Character DEFAULT_ESCAPE_CHARACTER = null;
+    public static final Character DEFAULT_QUOTE_CHARACTER = null;
+    public static final String DEFAULT_NULL_LITERAL = null;
     public static final boolean DEFAULT_IGNORE_ERRORS = false;
     public static final String DEFAULT_CHARSET = "UTF-8";
+    public static final int DEFAULT_BYTE_ARRAY_STREAM_LENGTH = 2048;
```
Comment (EMsnap) [inline inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatConstants.java:48]: not used, suggest use idea basic code check plugin to check again

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatDeserializer.java [hunk 4]
```diff
@@ -17,13 +17,42 @@
...
  * The deserializer for a given format.
  */
-public interface TableFormatDeserializer extends FlatMapFunction<byte[], Row>, ResultTypeQueryable<Row> {
+public abstract class TableFormatDeserializer extends RichFlatMapFunction<byte[], Row>
+        implements
+            ResultTypeQueryable<Row> {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String FORMAT_METRIC_GROUP_NAME = "format";
```
Comment (EMsnap) [inline inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatDeserializer.java:36]: not used

Comment (baomingyu) [reply inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatDeserializer.java]: delete

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatConstants.java (hunk 3 above)
Comment (baomingyu) [reply inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/TableFormatConstants.java]: delete

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultTableFormatSerializer.java (hunk 2 above)
Comment (baomingyu) [reply inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultTableFormatSerializer.java]: done

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/util/LogCounter.java (hunk 1 above)
Comment (baomingyu) [reply inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/util/LogCounter.java]: done

**Review by luchunliang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

File: inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultDeserializationSchema.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import org.apache.flink.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * The base implementation of {@link DeserializationSchema}.
+ */
+public abstract class DefaultDeserializationSchema<T> implements DeserializationSchema<T> {
```
Comment (vernedeng) [inline inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultDeserializationSchema.java:32]: AbstractDeserializationSchema

Comment (baomingyu) [reply inlong-sort/sort-formats/format-base/src/main/java/org/apache/inlong/sort/formats/base/DefaultDeserializationSchema.java:32]: next time handle

**Pull request merged**